### PR TITLE
Fix invalid continue/bear maneuvers

### PR DESCRIPTION
**Pull request by dgearhart**
# Issue

This PR fixes invalid continue/bear maneuvers

## Example 1
image
``` diff
< BEFORE
< 1: Drive northeast on Belair Road/US 1.
< 2: Continue on Belair Road/US 1.
< 3: You have arrived at Belair Road.
> AFTER
> 1: Drive northeast on Belair Road/US 1.
> 2: You have arrived at Belair Road.
```
---

## Example 2
image
``` diff
< BEFORE
< 1: Drive east on Frankfurter Ring.
< 2: You have arrived at Frankfurter Ring.
> AFTER
> 1: Drive east on Frankfurter Ring.
> 2: Keep left to stay on Frankfurter Ring toward Salzburg/Nürnberg/Passau/Landshut.
> 3: You have arrived at Frankfurter Ring.
```
---

## Example 3
image
``` diff
< BEFORE
< 2: Continue on I 895 South/Harbor Tunnel Thruway.
> AFTER
> 2: Keep right to take exit 62 onto I 895 South/Harbor Tunnel Thruway toward Annapolis.
```
---

## Tasklist

 - [x] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

### CODE REVIEW

File: run_route_scripts/results/multi_diff_results.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
```
Comment (gknisely) [inline run_route_scripts/results/multi_diff_results.sh:2]: What does this script do?  Can we add a note at the top

File: src/odin/maneuversbuilder.cc [hunk 2]
```diff
@@ -1821,6 +1821,7 @@ bool ManeuversBuilder::CanManeuverIncludePrevEdge(Maneuver& maneuver, int node_i
   if (IsFork(node_index, prev_edge.get(), curr_edge.get()) ||
       IsPedestrianFork(node_index, prev_edge.get(), curr_edge.get())) {
     maneuver.set_fork(true);
+
```
Comment (gknisely) [inline src/odin/maneuversbuilder.cc:1824]: nit.  extra space

File: test/maneuversbuilder.cc [hunk 3]
```diff
@@ -1835,9 +1835,9 @@ TEST(Maneuversbuilder, TestPathSlightRightXSlightLeftIsIntersectingForwardEdge)
   node = path.add_node();
   edge = node->mutable_edge();
   PopulateEdge(edge, {{"Horace Greeley Road", 0}}, 0.205258, 30.000000,
-               valhalla::RoadClass::kResidential, 35, 19, 6, 12, TripLeg_Traversability_kBoth, 0, 0,
+               valhalla::RoadClass::kResidential, 45, 19, 6, 12, TripLeg_Traversability_kBoth, 0, 0,
                0, 0, 0, 0, 0, 0, 0, 0, {}, {}, {}, {}, TripLeg_TravelMode_kDrive);
-  PopulateIntersectingEdge(node->add_intersecting_edge(), 355, 0, 0, TripLeg_Traversability_kBoth,
+  PopulateIntersectingEdge(node->add_intersecting_edge(), 3, 0, 0, TripLeg_Traversability_kBoth,
```
Comment (danpaz) [inline test/maneuversbuilder.cc:1840]: Why did these change? Was it required to make the test continue to pass?

File: src/odin/maneuversbuilder.cc (hunk 2 above)
Comment (dgearhart) [reply src/odin/maneuversbuilder.cc]: fixed - thanks

File: test/maneuversbuilder.cc (hunk 3 above)
Comment (dgearhart) [reply test/maneuversbuilder.cc:1840]: The logic to determine if the current edge is the straightest was not correct - the logic and padding threshold is now based on the delta from straight - not just the delta between the current edge and intersecting edge. Therefore, this test needed to be updated to validate the intersecting forward edge

**Review by danpaz — DISMISSED** [review]

File: run_route_scripts/results/multi_diff_results.sh (hunk 1 above)
Comment (dgearhart) [reply run_route_scripts/results/multi_diff_results.sh:16]: added description

**Review by gknisely — APPROVED** [review]

**Pull request merged**